### PR TITLE
COST-807: Create the GCP cost report api.

### PR DESCRIPTION
**Pull request by myersCody**
Jira Issue: https://issues.redhat.com/browse/COST-807

This pr is the groundwork foundation for the gcp api. I decided to start with the cost api because it will be the easiest. The group by & filter options are: `["account", "region", "service", "project"]`

## How to test

The first step to testing the cost api endpoint is to create a real gcp source:
- https://github.com/project-koku/koku#run-gcp-scenario

After the source is created, the database will be populated with most up to date information for the account in your .env file.

After the report is downloaded, processed, and the summarization of the data has been complete, we can now check out our new endpoint:

#### One: Test the single group_by with wildcard & specific value
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*) 
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[project]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[region]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[region]=*)

Note: After you test with the wildcard option, choose a value from your cost list and test the single group by options.
Example:
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A)

#### Two: Test filter combinations with wildcard & specific value
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[account]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[account]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[project]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[service]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[service]=*)
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[region]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?filter[region]=*)

Note: After you test with the wildcard option, choose a value from your cost list and test filter on a specific value.

#### Three: Test group_by & filter combinations
Example:
- http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[account]=*&filter[service]=FBF2-FC68-171A

#### Four: Multiple group_by
Example:
- [http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A&group_by[project]=*](http://127.0.0.1:8000/api/cost-management/v1/reports/gcp/costs/?group_by[service]=FBF2-FC68-171A&group_by[project]=*)

### CODE REVIEW

File: koku/api/report/gcp/serializers.py [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    service = StringOrListField(child=serializers.CharField(), required=False)
+    region = StringOrListField(child=serializers.CharField(), required=False)
+    project = StringOrListField(child=serializers.CharField(), required=False)
+
+
+class GCPQueryParamSerializer(ParamSerializer):
+    """Serializer for handling GCP query parameters."""
+
+    # Tuples are (key, display_name)
+    DELTA_CHOICES = ("cost", "cost")
```
Comment (blentz) [inline koku/api/report/gcp/serializers.py:71]: This needs to be a tuple of tuples. See, for example, koku/api/report/aws/serializers.py

```suggestion
    DELTA_CHOICES = (("cost", "cost"),)
```

File: koku/koku/cache.py [hunk 2]
```diff
@@ -93,6 +94,9 @@ def invalidate_view_cache_for_tenant_and_source_type(schema_name, source_type):
         )
     elif source_type in (Provider.PROVIDER_AZURE, Provider.PROVIDER_AZURE_LOCAL):
         cache_key_prefixes = (AZURE_CACHE_PREFIX, OPENSHIFT_AZURE_CACHE_PREFIX, OPENSHIFT_ALL_CACHE_PREFIX)
+    elif source_type in (Provider.PROVIDER_GCP, Provider.PROVIDER_GCP_LOCAL):
+        # TODO: Covert this back to a tuple, after we get more than one element.
+        cache_key_prefixes = [GCP_CACHE_PREFIX]
```
Comment (blentz) [inline koku/koku/cache.py:99]: I think we can just make this a tuple and extend it or adjust it as needed.

```suggestion
        cache_key_prefixes = (GCP_CACHE_PREFIX,)
```

**Review by blentz — CHANGES_REQUESTED** [review]
There's a couple tiny niggling issues that I can see. Once those are fixed, this looks okay to merge.

File: koku/koku/cache.py (hunk 2 above)
Comment (myersCody) [reply koku/koku/cache.py:99]: ahhhhhhhh. Thanks for this @blentz, I was trying to figure out why indexing was different based on if a tuple has a single element in it or not. I was missing the comma.

Example:
```
>>> single_tup = ("Cody")
>>> for idx in single_tup:
...     print(idx)
...
C
o
d
y
>>> single_tup_comma = ("Cody",)
>>> for idx in single_tup_comma:
...     print(idx)
...
Cody
```
This is a better solution and I learned something new. 🙇

**Review by dccurtis — APPROVED** [review]

**Review by blentz — APPROVED** [review]

**Pull request merged**